### PR TITLE
uses different parsing logic for platform versions when dot is missing

### LINKED ISSUE #9842: NuGet should not infer dots platform version when parsing TargetFramework
Use NuGet to parse the following `TargetFramework`: `net5.0-android21`

**Expected:** Parsed `TargetFrameworkVersion` is `21` (or `21.0`, etc)

**Actual:** Parsed `TargetFrameworkVersion` is `2.1`

NuGet is probably using the same version parsing logic for the platform version as it does for the target framework version.  However, the target framework version parsing automatically inserts version separators, so that, for example, net451 parses to .NET Framework 4.5.1.

We (probably) can't change how the parsing works for the target framework for compatibility reasons (though we probably want to disallow `net50` in favor of `net5.0`, and eventually should consider what to do with `net10`).  But we shouldn't bring this logic forward to the platform versions.

### PR DESCRIPTION
**Pull request by zkat**
Fixes: https://github.com/NuGet/Home/issues/9842

### CODE REVIEW

**Comment by zkat** [comment]
oop. need to add the API :)

File: test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs [hunk 1]
```diff
@@ -228,6 +228,7 @@ public void NuGetFramework_ProfileName(string folder, string expected)
         [InlineData("netcoreapp3.0", ".NetCoreApp,Version=v3.0")]
         [InlineData("net5.0-android", "net5.0-android")]
         [InlineData("net5.0-android", "net5.0-android0.0")]
+        [InlineData("net5.0-android10.0", "net5.0-android10")]
```
Comment (nkolev92) [inline test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs:231]: This test looks weird to me. 
Especially when comapred to the below 3.

Comment (zkat) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs:231]: can you elaborate? :)

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs:231]: It doesn't seem to round trip. 

`net5.0-android10.0` to `net5.0-android10` in this test, while the below test cases do preserve the `0`. 

In `        [InlineData("net5.0-ios14.0", "net5.0-ios14.0")]`, it's not removing the trailing 0.

Comment (zkat) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs:231]: you're reading it the wrong way around, I think. The one on the right is the input, and the left is the expected

Comment (zkat) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs:231]: You're reading it the other way around, I think? The expected output is on the left, and the second item is the input. Which... yes, is kinda confusing? It's not -supposed- to drop a .0

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs:231]: Yeah I was reading it the other way around. 

I know xunit asserts have the `expected,actual` ordering, but when passing in inline parametrs, the ordering is *usually*, `input1, input2, output`. 

It's a crazy world :D

**Review by nkolev92 — APPROVED** [review]
LGTM. 

I think this will lead to the folders inside the nupkgs to be structured as "net5.0-android21.0` but I think that's expected.

**Pull request merged**